src/arch/linux/mapping/mapping.c

```diff
@@ -178,8 +178,6 @@ static unsigned char *MEM_BASE32x(dosaddr_t a, int base)
 {
   if (mem_bases[base] == MAP_FAILED)
     return MAP_FAILED;
-  if (base == MEM_BASE || a >= ALIAS_SIZE)
-    return MEM_BASE32(a);
```

Comment:
Actually I think you just need to
enlarge an alias size for kvm_base,
but leave this check alone.
Then you won't need to use virtual2()
explicitly because it will fall-back to
mem_base at the certain address.

Reply:
> it's also confusing as
> MEM_BASE32x(0x110000, KVM_BASE) without any change returns &mem_base[0x110000] but I would 
> expect it to return &kvm_base[0x110000] naturally.

If you enlarge the alias size, then it
will return "&kvm_base[0x110000] naturally",
but not for the high mappings of ext_mem/xms.
I am not forcing that approach, its just to consider.